test/test_functions.jl

```diff
@@ -0,0 +1,54 @@
+# Various functions that are useful for running the tests, and used across several test sets.
+
+# Fetches the (required) Random package.
+using Random
+
+### Initial Condition/Parameter Generators ###
+
+# Generates a random initial condition (in the form of a map). Each value is a Float64.
+function rnd_u0(sys, rng; factor = 1.0, min = 0.0)
+    return [u => min + factor * rand(rng) for u in unknowns(sys)]
+end
+
+# Generates a random initial condition (in the form of a map). Each value is a Int64.
+function rnd_u0_Int64(sys, rng; n = 5, min = 0)
+    return [u => min + rand(rng, 1:n) for u in unknowns(sys)]
+end
+
+# Generates a random parameter set (in the form of a map). Each value is a Float64.
+function rnd_ps(sys, rng; factor = 1.0, min = 0.0)
+    return [p => min + factor * rand(rng) for p in parameters(sys)]
+end
+
+# Generates a random parameter set (in the form of a map). Each value is a Float64.
+function rnd_ps_Int64(sys, rng; n = 5, min = 0)
+    return [p => min + rand(rng, 1:n) for p in parameters(sys)]
+end
+
+# Used to convert a generated initial condition/parameter set to a vector that can be used for normal
+# DiffEq functions (that are created for manual comparison). Requires order list of symbols.
+function map_to_vec(map, syms)
+    syms_dict = Dict([ModelingToolkit.getname(entry[1]) => entry[2] for entry in map])
+    issetequal(keys(syms_dict), syms) || error("Map symbols ($(keys(syms_dict))) and symbol vector symbols ($(syms)) do not match.")
+    return [syms_dict[sym] for sym in syms]
+end
+
+### System function evaluation ###
+
+# Evaluates the the drift function of the ODE corresponding to a reaction network.
+function f_eval(rs::ReactionSystem, u, p, τ)
+    prob = ODEProblem(rs, u, (0.0, 0.0), p)
+    return prob.f(prob.u0, prob.p, τ)
```

Comment:
Technically, the order can affect stuff here. But until Chris decides that this is something he actually want to support there is not much to do. Generally, for variables, this is really not much of a problem, probably won't be in the future, but hopefully some improvement is made soon-ish.